### PR TITLE
Rewrite `tenzir-ctl import/export` into pipelines

### PR DESCRIPTION
**Pull request by jachris**
This changes `tenzir-ctl import json`, `tenzir-ctl import suricata` and `tenzir-ctl export json` to go through the normal `import` and `export` operators.

### CODE REVIEW

File: tenzir/integration/reference/export-pipeline-operator-parsing-after-expression/step_07.ref [hunk 1]
```diff
@@ -1,3 +1,3 @@
-{"flow_id_hashed": "1327c38b958ada2c", "src_ip": "83.16.53.68", "src_port": 1181, "dest_ip": "185.200.66.246"}
-{"flow_id_hashed": "30498b30eea22437", "src_ip": "83.16.53.68", "src_port": 138, "dest_ip": "83.16.53.56"}
-{"flow_id_hashed": "30498b30eea22437", "src_ip": "83.16.53.68", "src_port": 138, "dest_ip": "83.16.53.56"}
+{"flow_id_hashed": "1327c38b958ada2c", "src_ip": "80.24.82.106", "src_port": 1181, "dest_ip": "182.104.140.196"}
+{"flow_id_hashed": "30498b30eea22437", "src_ip": "80.24.82.106", "src_port": 138, "dest_ip": "80.24.82.24"}
+{"flow_id_hashed": "30498b30eea22437", "src_ip": "80.24.82.106", "src_port": 138, "dest_ip": "80.24.82.24"}
```
Comment (jachris) [inline tenzir/integration/reference/export-pipeline-operator-parsing-after-expression/step_07.ref:3]: Any idea why the output of `pseudonymize` changed here?

Comment (jachris) [reply tenzir/integration/reference/export-pipeline-operator-parsing-after-expression/step_07.ref:3]: Got it: https://github.com/tenzir/tenzir/pull/3787

File: tenzir/integration/tests.yaml [hunk 2]
```diff
@@ -263,32 +265,36 @@ tests:
...
 
-  Node json zeek conn:
-    tags: [node, import-export, zeek, json]
-    steps:
-      - command: -N import -s @./misc/schema/zeek-conn.schema -t zeek.conn.custom json
-        input: data/json/conn.log.json.gz
-      - command: -N export ascii 'where duration > 6s'
-      - command: -N export ascii 'where :timestamp >= 2011-08-15T03:48'
-      - command: -N export ascii 'where :time >= 2011-08-15T03:48'
+  # This test was broken by https://github.com/tenzir/tenzir/pull/3782.
```
Comment (dominiklohmann) [inline tenzir/integration/tests.yaml:268]: Just delete it alongside all the reference files. Doesn't make sense to keep it around.

File: tenzir/integration/tests.yaml [hunk 3]
```diff
@@ -330,24 +336,29 @@ tests:
...
       - command: count "net.src.ip == 192.168.168.100"
 
-  Arrow Full Data Model:
-    tags: [export, arrow]
-    steps:
-      - command: -N import --batch-encoding=arrow --schema-file="@./misc/schema/all-types.schema" --type=all_types json
-        input: data/json/all-types.json
-      - command: '-N export arrow "where #schema == \"all_types\""'
-        transformation: python @./misc/scripts/print-arrow.py
+  # This test was broken by https://github.com/tenzir/tenzir/pull/3782.
```
Comment (dominiklohmann) [inline tenzir/integration/tests.yaml:339]: Same here, just delete it alongside the reference files.

File: tenzir/integration/reference/export-pipeline-operator-parsing-after-expression/step_07.ref [hunk 4]
```diff
@@ -1,3 +1,3 @@
-{"flow_id_hashed": "1327c38b958ada2c", "src_ip": "83.16.53.68", "src_port": 1181, "dest_ip": "185.200.66.246"}
-{"flow_id_hashed": "30498b30eea22437", "src_ip": "83.16.53.68", "src_port": 138, "dest_ip": "83.16.53.56"}
-{"flow_id_hashed": "30498b30eea22437", "src_ip": "83.16.53.68", "src_port": 138, "dest_ip": "83.16.53.56"}
+{"flow_id_hashed": "1327c38b958ada2c", "src_ip": "80.24.82.106", "src_port": 1181, "dest_ip": "182.104.140.196"}
```
Comment (dominiklohmann) [inline tenzir/integration/reference/export-pipeline-operator-parsing-after-expression/step_07.ref:1]: Needs to be updated after #3787 is merged.

File: libtenzir/src/import_command.cpp
Comment (dominiklohmann) [inline libtenzir/src/import_command.cpp:1]: (Doesn't belong to any file really, just to have something to mark as resolved)

Please remove all *-writer and *-reader metrics from https://docs.tenzir.com/metrics/legacy_metrics

**Review by dominiklohmann — APPROVED** [review]

File: libtenzir/src/import_command.cpp
Comment (jachris) [reply libtenzir/src/import_command.cpp:1]: We agreed to not do this yet.

File: tenzir/integration/tests.yaml [hunk 5]
```diff
@@ -586,10 +583,12 @@ tests:
 
   Process Query For Field With Skip Attribute:
     tags: [import, export]
+    fixture: ServerTester
     steps:
-      - command: -N import --schema-file="@./misc/schema/zeek-with-skip.schema" zeek-json
+      - command: import -b --schema-file="@./misc/schema/zeek-with-skip.schema" zeek-json
```
Comment (lava) [inline tenzir/integration/tests.yaml:588]: Is this instance of `--schema-file` a leftover or intentional?

Comment (jachris) [reply tenzir/integration/tests.yaml:588]: `zeek-json` (and all other formats not explicitly listed in the PR description) are not changed here.

File: libtenzir/include/tenzir/exec_pipeline.hpp [hunk 6]
```diff
@@ -0,0 +1,30 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2023 The Tenzir Contributors
```
Comment (dominiklohmann) [inline libtenzir/include/tenzir/exec_pipeline.hpp:6]: Nit: It's 2024 now.

File: tenzir/integration/tests.yaml (hunk 5 above)
Comment (lava) [reply tenzir/integration/tests.yaml:588]: But `--schema-file` is an option to `import`, not for the specific format. What happens if we specify `import --schema-file=foo json`, is it silently ignored?

Comment (jachris) [reply tenzir/integration/tests.yaml:588]: Yes. For `json`, it is ignored, for `zeek-json` it is not ignored.

(And I think that's okay, given that the whole command is deprecated and to-be-removed soon)

**Pull request merged**